src/Adapter/Ftp.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Bolt\Filesystem\Adapter;
+
+use League\Flysystem\Adapter\Ftp as FtpAdapter;
+use League\Flysystem\Config;
+use Symfony\Component\Debug\Exception\ContextErrorException;
+
+/**
+ * FTP adapter.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+class Ftp extends FtpAdapter
+{
+    /** @var int */
+    protected $filePerm = 0664;
+    /** @var int */
+    protected $directoryPerm = 0775;
```

Comment:
OK, starting to get a picture of 6 months ago … A lot of running around in circles. :rofl: 

Upstream uses [`$directoryPerm`](https://github.com/thephpleague/flysystem-sftp/blob/master/src/SftpAdapter.php#L64)